Dockerfile

```diff
@@ -1,36 +1,51 @@
-FROM mcr.microsoft.com/dotnet/aspnet:8.0
-LABEL product="Technitium DNS Server"
-LABEL vendor="Technitium"
-LABEL email="support@technitium.com"
-LABEL project_url="https://technitium.com/dns/"
-LABEL github_url="https://github.com/TechnitiumSoftware/DnsServer"
-
-WORKDIR /opt/technitium/dns/
-
-RUN apt update; apt install curl -y; \
-curl https://packages.microsoft.com/config/debian/12/packages-microsoft-prod.deb --output packages-microsoft-prod.deb; \
-dpkg -i packages-microsoft-prod.deb; \
-rm packages-microsoft-prod.deb
-
-RUN apt update; apt install dnsutils libmsquic -y; apt clean -y;
-
-COPY ./DnsServerApp/bin/Release/publish/ .
-
-EXPOSE 5380/tcp
-EXPOSE 53443/tcp
-EXPOSE 53/udp
-EXPOSE 53/tcp
-EXPOSE 853/udp
-EXPOSE 853/tcp
-EXPOSE 443/udp
-EXPOSE 443/tcp
-EXPOSE 80/tcp
-EXPOSE 8053/tcp
-EXPOSE 67/udp
-
-VOLUME ["/etc/dns"]
+# syntax=docker.io/docker/dockerfile:1
 
+FROM mcr.microsoft.com/dotnet/aspnet:8.0
+RUN <<HEREDOC
+  # Add the MS repo to install `libmsquic` to support DNS-over-QUIC:
+  apt update && apt install -y curl
+  curl https://packages.microsoft.com/config/debian/12/packages-microsoft-prod.deb --output packages-microsoft-prod.deb
+  dpkg -i packages-microsoft-prod.deb
+  rm packages-microsoft-prod.deb
+  # `dnsutils` added to include the `dig` command for troubleshooting:
+  apt update && apt install -y libmsquic dnsutils
+  apt clean -y && rm -rf /var/lib/apt/lists/*
```

Comment:
With this base image you don't need to bring in `curl` / `ca-certificates`, the same can be accomplished with `ADD` (_the deb itself will be cached as a separate stage/layer, but only a minor benefit since it's so small anyway_).

```suggestion
# Add the MS repo to install `libmsquic` to support DNS-over-QUIC:
ADD --link https://packages.microsoft.com/config/debian/12/packages-microsoft-prod.deb /
RUN <<HEREDOC
  dpkg -i packages-microsoft-prod.deb && rm packages-microsoft-prod.deb
  # `dnsutils` added to include the `dig` command for troubleshooting:
  apt update && apt install -y libmsquic dnsutils
  apt clean -y && rm -rf /var/lib/apt/lists/*
```

Reply:
From hadolint: `DL3027 warning: Do not use apt as it is meant to be a end-user tool, use apt-get or apt-cache instead`

Reply:
Suggestion updated, I'll apply it since there was no discussion about preference.